agents-api/agents_api/web.py

```diff
@@ -89,6 +91,11 @@ def register_exceptions(app: FastAPI) -> None:
     max_age=3600,
 )
 
+app.add_middleware(GZipMiddleware, minimum_size=1000, compresslevel=3)
```

Comment:
The Authorization logic should be moved into a middleware and executed before any parsing begins